### ISSUE TITLE
Extended styles of a styled-component not rendering with renderToNodeStream

### ISSUE THREAD

**Issue by mikeheddes** [styled-components/styled-components#2020]
### System:
 - OS: macOS High Sierra 10.13.6
 - CPU: x64 Intel(R) Core(TM) i7-7660U CPU @ 2.50GHz
 - Memory: 403.07 MB / 8.00 GB
 - Shell: 3.2.57 - /bin/bash
### Binaries:
 - Node: 8.11.1 - /usr/local/bin/node
 - Yarn: 1.9.4 - ~/.yarn/bin/yarn
 - npm: 6.4.0 - /usr/local/bin/npm
### npmPackages:
 - babel-plugin-styled-components: ^1.7.1 => 1.7.1
 - styled-components: ^4.0.0-beta.7 => 4.0.0-beta.7

## Reproduction
```js
// babel.config.js
const commonPlugins = [
  '@babel/proposal-class-properties',
  '@babel/proposal-export-default-from',
  '@babel/proposal-export-namespace-from',
  '@babel/syntax-dynamic-import',
]

module.exports = {
  presets: [['@babel/env', { modules: false }], '@babel/react'],
  env: {
    production: {
      plugins: [
        ['styled-components', { displayName: false }],
        'react-loadable/babel',
        'transform-react-remove-prop-types',
        '@babel/transform-react-constant-elements',
        '@babel/transform-react-inline-elements',
      ].concat(commonPlugins),
    },
    development: {
      plugins: ['styled-components', 'react-loadable/babel'].concat(
        commonPlugins
      ),
    },
  },
}
```
I use `interleaveWithNodeStream` and `renderToNodeStream`. I updated from `styled-components@3.3.3` which was working. See my project repo for the setup https://github.com/mikeheddes/mikeheddes-website/tree/update-versions. In an other project I tried using renderToString which worked correctly. 

## Steps to reproduce
```
$ git clone https://github.com/mikeheddes/mikeheddes-website.git
$ cd mikeheddes-website
$ git checkout update-versions
$ yarn install
$ yarn build:dll
$ yarn start
```
Now go to http://localhost:3000/music/one_kiss and note that the footer is not at the bottom of the page on MacBook Pro 13" full screen. The code for this page can be found in `src/containers/MusicItem/index.jsx`. See line 27 - 29, the Main component. The extended styles are not rendered.

## Expected Behavior
This is what I tried to get in my project https://codesandbox.io/s/yp3n137781.

## Actual Behavior
The styles of Foo are rendered but the extended styles of Bar are not rendered.

Hope this helps.

**Comment by probablyup** [comment]
Can you try

```
npm install styled-components@4.0.0-beta.7-hoist
```

and let me know if that fixes it?

**Comment by mikeheddes** [comment]
Yes, I just did and it works! Thanks

**Comment by probablyup** [comment]
Awesome! Going to wait for a few more responses and probably will merge it tomorrow

**Comment by probablyup** [comment]
Released in [v4.0.0-beta.8](https://github.com/styled-components/styled-components/releases/tag/v4.0.0-beta.8)
